cpp/src/io/json/json_column.cu

```diff
@@ -576,11 +602,24 @@ table_with_metadata device_parse_nested_json(device_span<SymbolT const> d_input,
   std::vector<column_name_info> out_column_names;
   auto parse_opt = parsing_options(options, stream);
 
-  // Iterate over the struct's child columns and convert to cudf column
-  size_type column_index = 0;
-  for (auto const& col_name : root_struct_col.column_order) {
-    auto& json_col = root_struct_col.child_columns.find(col_name)->second;
+  bool const has_column_order =
+    options.is_enabled_prune_columns() and
+    std::holds_alternative<schema_element>(options.get_dtypes()) and
+    std::get<schema_element>(options.get_dtypes()).column_order.has_value() and
+    not std::get<schema_element>(options.get_dtypes()).column_order->empty();
```

Comment:
Can we throw if `options.is_enabled_prune_columns()` is `false` but column order is given? The users may want to specify column order but they may forget to enable prune column.

Reply:
It is possible, the user wants to prune column, but maintain the order as in json file. that's why that throw was not added.